### PR TITLE
Add Karma

### PR DESCRIPTION
**Pull request by nkbt**

### CODE REVIEW

File: test/filter-test.js [hunk 1]
```diff
@@ -9,8 +8,8 @@ describe('filter', () => {
 
     expect(result).toContain(1);
     expect(result).toContain(3);
-    expect(result).toNotContain(2);
-    expect(result).toNotContain(4);
+    expect(result).not.toContain(2);
```
Comment (nkbt) [inline test/filter-test.js]: !This was the only difference after replacing expect with jasmine

File: karma.conf.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        {type: 'text-summary', file: 'text-summary.txt'},
+      ],
+    },
+    captureTimeout: 90000,
+    browserNoActivityTimeout: 60000,
+    port: 9876,
+    colors: true,
+    logLevel: config.LOG_INFO,
+    autoWatch: false,
+    browsers: ['PhantomJS'],
```
Comment (nkbt) [inline karma.conf.js]: Only phantom.
If you wish to debug tests in browsers, simply run `npm run test:dev` and point your browser (any) to http://localhost:9876 (karma port).

**Comment by chenglou** [comment]
Ehh, does jasmine support good async testing like mocha's `done` passed into the callback? Because we're probably gonna need it.

**Comment by nkbt** [comment]
Sure, it does, but it is always better to use spies and mocked time instead.

```js
  it("should support async execution of test preparation and expectations", function(done) {
    value++;
    expect(value).toBeGreaterThan(0);
    done();
  });
```

**Comment by chenglou** [comment]
Cool, that's jasmine 2.0 right. It's been a while but I recall jasmine being on < 2.0 for quite a while.
Will mocked time work correctly with the new animation loop? Say, I want fine-grained simulation for when the next raf is. And I assume this works in phantom?

**Comment by nkbt** [comment]
Mocking works pretty well. Phantom, yes. I would not submit a PR that does
not work lol :)

File: package.json [hunk 3]
```diff
@@ -28,9 +34,10 @@
     "build-demos": "NODE_ENV=production webpack -p",
     "lint": "eslint --ext .js,.jsx .",
     "prerelease": "babel src --out-dir lib && NODE_ENV=production webpack --config webpack.prod.config.js",
-    "test": "mocha --compilers js:babel/register --recursive test",
-    "test:dev": "npm test -- --watch",
-    "test:cov": "`npm bin`/babel-node `npm bin`/isparta cover --report text --report html `npm bin`/_mocha -- --recursive src"
+    "karma": "karma start ./karma.conf.js",
+    "test": "npm run karma --  --single-run --reporters nyan",
+    "test:dev": "npm run karma -- --no-single-run --auto-watch --reporters nyan",
```
Comment (chenglou) [inline package.json]: I asked for a minimal PR. Is nyan cat minimal?!

jkjk =)

**Comment by chenglou** [comment]
:+1:

**Pull request merged**